File: 백준/Silver/1929. 소수 구하기/소수 구하기.cc
#include<iostream>
using namespace std;
int main(){
        int start,end;
        cin>>start>>end;

        bool *arr;
        arr =new bool[end+1];

        for(int i=2; i<=end; i++)
                arr[i]=true;

        for(int i=2; i<=end; i++){//i가 나누는 수
                if(arr[i]==false)
                        continue;
                for(int j= i+i; j<=end; j +=i){
                        arr[j]=false;
                }
        }

        for(int i=start; i<=end; i++){
                if(arr[i]==true)
                        cout<<i<<"\n";
        }
}

File: 백준/Silver/1929. 소수 구하기/소수 구하기.py
import sys

st, n = map(int, sys.stdin.readline().split())

def prime_numbers(start, end):
    arr = [True] * (end + 1)
    arr[0] = arr[1] = False  # 0과 1은 소수 아님

    for i in range(2, int(end ** 0.5) + 1):
        if arr[i]:
            for j in range(i * i, end + 1, i):
                arr[j] = False

    return [i for i in range(start, end + 1) if arr[i]]

result = '\n'.join(str(s) for s in prime_numbers(st, n))
sys.stdout.write(result + '\n')
